src/datarecord.cpp

```diff
@@ -582,3 +582,32 @@ void datarecord::schedule(std::vector<rec_ptr>& thisi, double maxtime,
     thisi.push_back(evon);
   }
 }  
+
+/* 
+ * Inserts a record at the earliest or latest opportunity based on time. 
+ * Boolean flag indicates if the record should get inserted before or after all 
+ * other records at the time of that event. 
+ * 
+ * This is currently being used to insert records for lag times as well as
+ * infusion end. 
+ * 
+ */
+void insert_record(reclist& thisi, const int start, rec_ptr& rec, 
```

Comment:
New with this PR.

Reply:
The if/else code is almost identical, just a matter if we have to exceed the target time or not (in case there is already an event at the target time).